src/Microsoft.AspNetCore.Mvc.Abstractions/Routing/AttributeRouteInfo.cs

```diff
@@ -36,5 +38,7 @@ public class AttributeRouteInfo
         /// Gets or sets a value that determines if the route entry associated with this model participates in path matching (inbound routing).
         /// </summary>
         public bool SuppressPathMatching { get; set; }
+
+        public IParameterTransformer RouteTokenTransformer { get; set; }
```

Comment:
So one issue that comes to mind here is that this will keep all of the transformer instance around in memory long after they have served their purpose, that's really not ideal.  

Do we really need this here or just on the `AttributeRouteModel`?

Reply:
Yeah it needs to be here because `AttributeRouteModel.ReplaceTokens` uses the values from `AttributeRouteInfo`.

Could set it to null after token replacement has run? And maybe also make it internal on AttributeRouteInfo?

Reply:
That's a good stop gap, but why couldn't this be passed as parameter to `ReplaceTokens`? What else would need to change?

Reply:
It is called from `ControllerActionDescriptorBuilder.Build`. It need to come somewhere from `ApplicationModel`.